### PR TITLE
geomfn: fix st_simplify with NaN

### LINKED ISSUE #63620: geomfn: st_simplify with NaN causes node to crash
This query causes the node to crash without leaving any logs behind.

```
SELECT st_simplify('0104000000020000000101000000AC885A457886F9C1C9EEF30F7308F3C101010000007411E55B3F66EDC10BBEFA0ABF45F0C1':::GEOMETRY, 'nan':::FLOAT8);
```

It works on PostGIS though

```
postgres@localhost:postgres> SELECT st_simplify('0104000000020000000101000000AC885A457886F9C1C9EEF30F7308F3C101010000007411E55B3F66EDC10BBEFA0ABF45F0C1'::GEOMETRY, 'NaN'::FLOAT8)
+--------------------------------------------------------------------------------------------------------+
| st_simplify                                                                                            |
|--------------------------------------------------------------------------------------------------------|
| 0104000000020000000101000000AC885A457886F9C1C9EEF30F7308F3C101010000007411E55B3F66EDC10BBEFA0ABF45F0C1 |
+--------------------------------------------------------------------------------------------------------+
SELECT 1

postgres@localhost:postgres> SELECT st_astext(st_simplify('0104000000020000000101000000AC885A457886F9C1C9EEF30F7308F3C101010000007411E55B3F66EDC10BBEFA0ABF45F0C1'::GEOMETRY,
 'NaN'::FLOAT8))
+-----------------------------------------------------------------------------------+
| st_astext                                                                         |
|-----------------------------------------------------------------------------------|
| MULTIPOINT(-6851888213.65837 -5109133567.2458,-3945921247.15838 -4368101551.6714) |
+-----------------------------------------------------------------------------------+
SELECT 1
```

cc @otan @andyyang890

### PR DESCRIPTION
**Pull request by techytoes**
Release note (SQL Change): Earlier st_simplify with NaN caused node to crash.
This fixes the behaviour to align it more to PostGIS.

Fixes: #63620

### CODE REVIEW

File: pkg/geo/geomfn/topology_operations.go [hunk 1]
```diff
@@ -96,6 +98,9 @@ func Difference(a, b geo.Geometry) (geo.Geometry, error) {
 
 // Simplify returns a simplified Geometry.
 func Simplify(g geo.Geometry, tolerance float64) (geo.Geometry, error) {
+	if math.IsNaN(tolerance) {
```
Comment (otan) [inline pkg/geo/geomfn/topology_operations.go:101]: can you change this to `if math.IsNaN(tolerance) g.ShapeType2D() == geopb.ShapeType_POINT || g.ShapeType2D() == geopb.ShapeType_MULTIPOINT `, and add tests for points and multipoints?

Comment (techytoes) [reply pkg/geo/geomfn/topology_operations.go]: I have made the changes. please review this again.

Thanks🙇🏻‍♂️

**Review by otan — APPROVED** [review]
looks good! thanks!

bors r+

**Comment by techytoes** [comment]
TFTR! @otan 🙇🏻‍♂️

**Pull request merged**